cime/src/share/util/shr_reprosum_mod.F90

```diff
@@ -920,8 +1016,10 @@ subroutine shr_reprosum_int (arr, arr_gsum, nsummands, dsummands, nflds, &
 
           max_error(ifld,ithread) = 0
           not_exact(ifld,ithread) = 0
-
           i8_arr_tlsum_level(:,ifld,ithread) = 0_i8
+
+          if (skip_field(ifld)) cycle
+
```

Comment:
Move this `cycle` 5 lines higher to occur right after `do ...` for a quicker cycle to the next field?

Reply:
We need the statements between the do ... and this location executed in order not to have 

    if (skip_field(ifld)) cycle

inserted in subsequent do loops.